src/index.js

```diff
@@ -114,21 +149,26 @@ class CID {
     }
 
     // otherwise, assemble the CID from the parameters
+    // type checker will not accept `version`, `codec`, `multihash` as is
+    // because types don't correspond to [number, string, Buffer] so we
+    // use any below as we know `CID.validateCID` will throw if types do not
+    // match up.
+    const [$version, $codec, $multihash]/*: any */ = [version, codec, multihash]
 
     /**
      * @type {number}
      */
-    this.version = version
+    this.version = $version
```

Comment:
Would flow automatically pick the type from the JSDoc? Or where does it know that `this.version` is a `number`?

Reply:
Comment above should clarify this

Reply:
Oh sorry I’ve misread this: see line 78 where types of fields are declared